Tests/AuthTests/ExtractParamsTests.swift

```diff
@@ -13,13 +13,20 @@ final class ExtractParamsTests: XCTestCase {
     let code = UUID().uuidString
     let url = URL(string: "io.supabase.flutterquickstart://login-callback/?code=\(code)")!
     let params = extractParams(from: url)
-    XCTAssertEqual(params, [Params(name: "code", value: code)])
+    XCTAssertEqual(params, ["code": code])
   }
 
   func testExtractParamsInFragment() {
     let code = UUID().uuidString
     let url = URL(string: "io.supabase.flutterquickstart://login-callback/#code=\(code)")!
     let params = extractParams(from: url)
-    XCTAssertEqual(params, [Params(name: "code", value: code)])
+    XCTAssertEqual(params, ["code": code])
+  }
+
+  func testExtractParamsInBothFragmentAndQuery() {
```

Comment:
Quick check on my understanding here: if we have both a query param and a hash param with `code` does the query param take predence?  Likewise for `error` 

(e.g.  currently does "io.supabase.flutterquickstart://login-callback/?code=123#code=abc" result in 
["code": 123 ]) 

I think with the JS lib we had a few issues related to terms which appeared in both the query and hash fragments so might be worth  confirming  against the behaviour there before merging